internal/sponge/item/model.go

```diff
@@ -32,3 +39,19 @@ func (item *Item) Validate() error {
 
 	return nil
 }
+
+// InferIDFromData infers an item_id from type and source id.
+func (item *Item) InferIDFromData() error {
+
+	// Does the source_id field exist in the data?
+	// If a source id value is found, use it to compose the item_id for this item.
+	idValue, ok := item.Data[sourceIDField]
+	if !ok {
+		return fmt.Errorf("Cannot Infer ID: Unable to find source id field")
+	}
+
+	item.ID = item.Type + "_" + fmt.Sprintf("%v", idValue)
```

Comment:
Good call. This allowed me to create a const for the template, which is much more readable. Updated everywhere.